### PR TITLE
Test and build wheels on 3.11

### LINKED ISSUE #2177: Python 3.11 wheels
It would be great to have wheels for Python 3.11

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #2177

### CODE REVIEW

**Comment by benjeffery** [comment]
Ah, got it. Should be all good now.

File: msprime/ancestry.py [hunk 1]
```diff
@@ -1510,7 +1510,7 @@ def _run_until(self, end_time, event_chunk=None, debug_func=None):
                     "Please open an issue on GitHub"
                 )
             logger.debug(
-                "time=%g ancestors=%d ret=%s", self.time, self.num_ancestors, ret
+                f"time={self.time} ancestors={self.num_ancestors} ret={repr(ret)}"
```
Comment (jeromekelleher) [inline msprime/ancestry.py:1513]: Why this change? Are these logging converters not supported after 3.11? They only generate the string as needed with log level, so are useful

**Review by jeromekelleher — COMMENTED** [review]
Lgtm - logging changes seem like a regression though?

File: msprime/ancestry.py (hunk 1 above)
Comment (benjeffery) [reply msprime/ancestry.py:1513]: In Py3.11 `IntEnum`'s  `__str__` method has reverted to that of a normal int. The other way to fix this is to change the checked for text to test for an int instead of the string. I'll change it to that.

File: tests/test_ancestry.py [hunk 2]
```diff
@@ -764,7 +765,11 @@ def test_debug_logging_dtwf(self, caplog):
                 " until max time: inf"
             )
             assert caplog.messages[-1].startswith("Completed at time")
-            assert caplog.messages[2] == "time=1 ancestors=3 ret=ExitReason.MAX_EVENTS"
+            print("WOP", caplog.messages[2])
```
Comment (jeromekelleher) [inline tests/test_ancestry.py:768]: Stray print

File: tests/test_ancestry.py [hunk 3]
```diff
@@ -764,7 +765,11 @@ def test_debug_logging_dtwf(self, caplog):
...
+            assert (
+                caplog.messages[2] == "time=1 ancestors=3 ret=ExitReason.MAX_EVENTS"
```
Comment (jeromekelleher) [inline tests/test_ancestry.py:770]: Just change to starts with the prefix that's shared

**Review by jeromekelleher — APPROVED** [review]
Looks good, ready to merge after cleanup

**Comment by benjeffery** [comment]
@mergify rebase
